core/server/worker/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -188,8 +188,10 @@ private InputStream openAtPosition(long position) throws IOException {
       if (mStream == null) {
         UnderFileSystem ufs = UnderFileSystem.Factory.get(mUri);
         mStream = ufs.open(mUri, OpenOptions.defaults().setOffset(position));
-      } else {
+      } else if (mStream instanceof Seekable) {
         ((Seekable) mStream).seek(position);
+      } else {
+        throw new IOException("Unsupported operation.");
```

Comment:
Could you add some detail to this message (ie. `Under File System stream does not implement seek`)? Though I think if it goes into this code block the implementation is broken. Perhaps we should enforce `mStream` is seekable? I'm fine with leaving this for another PR.